src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -0,0 +1,176 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.controller;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.UUID;
+
+import codeu.model.data.User;
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+
+import java.util.List;
+import java.util.ArrayList;
+import codeu.model.store.basic.UserStore;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class AdminServletTest {
+
+  private AdminServlet adminServlet;
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private RequestDispatcher mockRequestDispatcher;
+  private UserStore mockUserStore;
+  private ConversationStore mockConvoStore;
+  private MessageStore mockMessageStore;
+
+  @Before
+  public void setup() {
+    adminServlet = new AdminServlet();
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+
+    mockUserStore = Mockito.mock(UserStore.class);
+    mockConvoStore = Mockito.mock(ConversationStore.class);
+    mockMessageStore = Mockito.mock(MessageStore.class);
+
+    User fakeUser =
+        new User(
+            UUID.randomUUID(),
+            "test username",
+            "$2a$10$bBiLUAVmUFK6Iwg5rmpBUOIBW6rIMhU1eKfi3KR60V9UXaYTwPfHy",
+            Instant.now());
```

Comment:
Just a question, what is the Instant.now() used for?

Reply:
So I had the exact same question, but if you look inside the User class in the path main/java/codeu/model/data, you can see that a User consists of 4 parameters which are: an ID,a name, a password hash, an a parameter named "creation." My guess is that creation stores the time the user object was made, so passing Instant.now() for that parameter gives it the current time from your system clock.

Reply:
Instant.now is just a time stamp of when the user was created. I am using it here I think just because it is necessary as per the constructor of the User but its not really useful outside of that.

Actually yes it is useful. I am going to use that to implement a "recently registered users" type of statistic for the admin page. Great suggestion haha thanks!